### ISSUE TITLE
Commit 7f99eb7a48e8e95cc3059eda7910cd6aaf7341cc broke flyspell with markdown

### ISSUE THREAD

**Issue by RoryO** [hlissner/doom-emacs#4407]
**What did you expect to happen?**
Flyspell with markdown-mode provides inline spelling hints

**What actually happened?**
Flyspell after commit 7f99eb7a48e8e95cc3059eda7910cd6aaf7341cc does not check the document inline. Manual checking works.

**Additional details:**
before 7f99eb7a48e8e95cc3059eda7910cd6aaf7341cc
![a](https://user-images.githubusercontent.com/19547/101833740-463d9300-3aee-11eb-89d3-58c646017761.gif)

after 7f99eb7a48e8e95cc3059eda7910cd6aaf7341cc
![b](https://user-images.githubusercontent.com/19547/101833755-4b024700-3aee-11eb-9b81-c1137720eb82.gif)

<details><pre>
If an error message is involved include a backtrace of it.

How to acquire a backtrace:
https://github.com/hlissner/doom-emacs/blob/develop/docs/getting_started.org#how-to-extract-a-backtrace-from-an-error
</pre></details>

**Steps to reproduce:**
1. Create a buffer with markdown-mode
2. Misspell something

**System information:**
<details><pre>
```
SYSTEM  type       windows-nt
        config     x86_64-w64-mingw32
        shell      A:/emacs27/libexec/emacs/27.1/x86_64-w64-mingw32/cmdproxy.exe
        uname      n/a
        path       (C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.26.28801\bin\HostX64\x64 C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\IDE\CommonExtensions\Microsoft\TestWindow C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\MSBuild\Current\bin\Roslyn C:\Program Files (x86)\Windows Kits\10\bin\10.0.18362.0\x64 C:\Program Files (x86)\Windows Kits\10\bin\x64 C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\\MSBuild\Current\Bin C:\Windows\Microsoft.NET\Framework64\v4.0.30319 C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\IDE\ C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\Tools\ C:\Program Files\PowerShell\7 C:\VulkanSDK\1.2.154.1\Bin C:\VulkanSDK\1.2.135.0\Bin C:\VulkanSDK\1.1.130.0\Bin C:\WINDOWS\system32 C:\WINDOWS C:\WINDOWS\System32\Wbem C:\WINDOWS\System32\WindowsPowerShell\v1.0\ C:\WINDOWS\System32\OpenSSH\ C:\ProgramData\chocolatey\bin C:\Program Files\PowerShell\7-preview\preview C:\Program Files\PowerShell\7\ C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit\ C:\Program Files\dotnet\ C:\Users\$USER\scoop\apps\python\current\Scripts C:\Users\$USER\scoop\apps\python\current C:\Users\$USER\scoop\apps\nodejs\current\bin C:\Users\$USER\scoop\apps\nodejs\current C:\Users\$USER\scoop\apps\elixir\current\bin C:\Users\$USER\scoop\apps\ruby\current\gems\bin C:\Users\$USER\scoop\apps\ruby\current\bin C:\Users\$USER\scoop\shims C:\Program Files\CMake\bin C:\msys64\mingw64\bin\ C:\msys64\usr\local\bin C:\msys64\usr\bin C:\Users\$USER\AppData\Local\Programs\Python\Python38-32\Scripts\ C:\Users\$USER\AppData\Local\Programs\Python\Python38-32\ C:\Users\$USER\AppData\Local\Microsoft\WindowsApps C:\Users\$USER\AppData\Local\Programs\Microsoft VS Code\bin a:\bin a:\emacs27\bin C:\Program Files\LLVM\bin C:\Users\$USER\AppData\Local\Keybase\ C:\Users\$USER\AppData\Local\GitHubDesktop\bin C:\Users\$USER\elixir-ls C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\IDE\CommonExtensions\Microsoft\CMake\CMake\bin C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\Common7\IDE\CommonExtensions\Microsoft\CMake\Ninja a:/emacs27/libexec/emacs/27.1/x86_64-w64-mingw32/)
EMACS   dir        ~/.emacs.d/
        version    27.1
        build      Aug 21, 2020
        buildopts  --without-dbus --host=x86_64-w64-mingw32 --without-compress-install 'CFLAGS=-O2 -static'
        features   XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY W32NOTIFY ACL GNUTLS LIBXML2 HARFBUZZ ZLIB TOOLKIT_SCROLL_BARS MODULES THREADS JSON PDUMPER LCMS2 GMP
        traits     (gui server-running envvar-file)
DOOM    dir        a:/dotfiles/.doom.d/
        version    2.0.9
        font       #<font-spec nil nil Consolas nil nil nil nil nil 36 nil nil nil nil>
        theme      doom-material
        build      HEAD -> develop 35ae68550 2020-12-10 13:34:40 -0800
        elc-files  0
        modules    (:completion company ivy :ui doom doom-dashboard hl-todo modeline nav-flash ophints (popup +all +defaults) vc-gutter vi-tilde-fringe workspaces :editor (evil +everywhere) fold snippets :emacs dired electric ibuffer undo :checkers syntax (spell +flyspell +hunspell) grammar :tools (eval +overlay) lookup lsp :lang (cc +lsp) data emacs-lisp markdown (org +dragndrop +hugo +jupyter +pandoc +pomodoro +present) :config (default +bindings +smartparens))
        packages   ((writeroom-mode) (zig-mode) (evil-markdown :disable t))
        unpin      (n/a)
        elpa       (n/a)
```

</pre></details>

**Comment by hlissner** [comment]
As of 2e61fbb this issue should be resolved. Let me know if that isn't the case and I'll reopen it. Thanks for bringing it to my attention!

**Comment by RoryO** [comment]
Fixed after 2e61fbb. Thank you!
